compiler/src/dmd/expressionsem.d

```diff
@@ -10843,22 +10856,9 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                                 return;
                             }
                         }
-                        else if (sd.hasMoveCtor && !e2x.isCallExp() && !e2x.isStructLiteralExp())
+                        else if (sd.hasMoveCtor && !e2x.isStructLiteralExp())
                         {
                             // #move
-                            /* The !e2x.isCallExp() is because it is already an rvalue
-                               and the move constructor is unnecessary:
```

Comment:
@WalterBright I had to change this line, since it explicitly rejected rvalue call expressions.
Can you explain this case? Why did you assume call expressions are already an rvalue? Calls can return ref, in which case, they are not an rvalue...

Does this expression need different logic so it only lets in this particular case, or will this work generally?

Reply:
I suspect this change is what broke the `rvalue2.d` unit test, but I don't understand why...

Reply:
Don't worry, I fixed it...